src/network/stream_reader.rs

```diff
@@ -28,26 +28,26 @@ use network::message::RawNetworkMessage;
 use consensus::encode;
 
 /// Struct used to configure stream reader function
-pub struct StreamReader<'a> {
+pub struct StreamReader<'a, R: 'a> {
     /// Stream to read from
-    pub stream: &'a mut Read,
+    pub stream: &'a mut R,
```

Comment:
Why is this a reference?

Reply:
Good question ha. I just changed the trait without giving it too much thought, i'll change it